README-Docker.md

```diff
@@ -0,0 +1,248 @@
+
+This example uses a Mac M1/M2 (arm64) but you can specify the platform and architecture as parameters when building the image and running the container.  
+
+The steps would be the same on an amd64 machine with NVIDIA GPUs as for an arm64 Mac; the platform is configurable. You [can't access the Mac M1/M2 GPU cores from Docker containers](https://github.com/pytorch/pytorch/issues/81224) and performance is reduced compared with running it directly on macOS but for development purposes it's fine. Once you're done with development tasks on your laptop you can build for the target platform and architecture and deploy to an environment with NVIDIA GPUs on-premises or in the cloud.
+
+### Prerequisites
+[Install Docker](https://gist.github.com/santisbon/2165fd1c9aaa1f7974f424535d3756f7#docker)  
+On the Docker Desktop app, go to Preferences, Resources, Advanced. Increase the CPUs and Memory to avoid this [Issue](https://github.com/lstein/stable-diffusion/issues/342). You may need to increase Swap and Disk image size too.  
+
+Create a Docker volume for the downloaded model file
+```
+docker volume create my-vol
+```
+
+Copy the model file (we'll need it at run time) to the Docker volume using a lightweight Linux container. You just need to create the container with the mountpoint; no need to run it.
+```Shell
+docker create --platform linux/arm64 --name dummy --mount source=my-vol,target=/data alpine 
+
+cd ~/Downloads # or wherever you saved sd-v1-4.ckpt
+docker cp sd-v1-4.ckpt dummy:/data
+```
+
+### Setup
+Set the fork you want to use.  
+Download the Miniconda installer (we'll need it at build time). Replace the URL with the version matching your system.
+```Shell
+GITHUB_STABLE_DIFFUSION="https://github.com/santisbon/stable-diffusion.git"
+
+cd ~
+git clone $GITHUB_STABLE_DIFFUSION
+
+cd stable-diffusion/docker-build
+chmod +x entrypoint.sh
+
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-aarch64.sh -O anaconda.sh && chmod +x anaconda.sh
+```
+
+Build the Docker image. Give it any tag ```-t``` that you want.  
+Tip: Check that your shell session has the env variable set (above) with ```echo $GITHUB_STABLE_DIFFUSION```.  
+```condaarch``` will restrict the conda environment to the right architecture when installing packages. It can take on: ```linux-64```, ```osx-64```, ```osx-arm64```.
+```Shell
+docker build -t santisbon/stable-diffusion \
+--platform linux/arm64 \
+--build-arg condaarch="osx-arm64" \
+--build-arg gsd=$GITHUB_STABLE_DIFFUSION \
+--build-arg sdreq="requirements-linux-arm64.txt" \
+.
+```
+
+Run a container using your built image e.g.
+```Shell
+docker run -it \
+--rm \
+--platform linux/arm64 \
+--name stable-diffusion \
+--hostname stable-diffusion \
+--mount source=my-vol,target=/data \
+--expose 9090 \
+--publish 9090:9090 \
+santisbon/stable-diffusion
+```
+Tip: Make sure you've created the Docker volume (above)
+
+## Option B - Directly on Apple silicon
```

Comment:
We do NOT need more install instructions for bare-metal Apple silicon. We already have a specific README for this.

Reply:
I'll remove the section.